src/frontends/sql/decoder_context.cc

```diff
@@ -0,0 +1,36 @@
+#include "src/frontends/sql/decoder_context.h"
+
+namespace raksha::frontends::sql {
+
+using ir::NamedValueMap;
+using ir::Operation;
+using ir::Value;
+
+static NamedValueMap MakeNamedValueMapFromVector(absl::string_view prefix,
```

Comment:
Also, MLIR code base has a pattern that would help here and elsewhere. Basically create a class as follows:

```
class MergeOperator : public Operator {
public:
  std::unique_ptr<Operation>  MakeOperation(std::vector<Value> inputs, std::vector<Value> outputs);
  // The logic that direct inputs have  "direct_input_" prefix will be capture here. 
  ValueRange GetDirectInputs(const Operation& operation) {
   } 

};
```
Then, we can use it to create and manipulate relevant operations in a more straightforward way:

```
MergeOperator::CreateOperation(....);
...
MergeOperator::GetControlInputs(operation);
```

Not suggesting you make these changes in this PR, but seems like a useful mechanism to build.

Reply:
Interesting, that does seem useful! In MLIR, is this primarily used as a test helper or as an implementation thing? I think I might add this, but as a `testonly` class right now to prevent from having to write yet-more tests for it. If we ever decide that we want to use it in actual non-test code, we can unit test it then. Considering the MVP nature of the current work, I think that it's more likely to be stripped out entirely than to be moved to non-test status.

Reply:
I've created a class `MergeOperationView` that does something similar to this, but with composition rather than inheritance. In some ways, the inheritance-based solution would be better, but this prevents from having to modify the `BlockBuilder` and other classes that expect a regular `Operation`.